### PR TITLE
docs(roadmap): update with more detail

### PR DESCRIPTION
**Pull request by hammy2899**
## Description

Added more detail to the roadmap. If anyone thinks something is missing leave a comment and I will add it.

### CODE REVIEW

File: ROADMAP.md [hunk 1]
```diff
@@ -1,34 +1,84 @@
...
+
+### Library Implementations:
+
+This is a list of libraries we are wanting and proposing to create implementations for.
+
+- DOM (Standard JavaScript DOM) 🛳
+- React 🛳
+- Vue 🛳
+- Angular 💡
+- Meteror 💡
```
Comment (lubber-de) [inline ROADMAP.md]: Typo: Should read Meteor

File: ROADMAP.md [hunk 2]
```diff
@@ -1,34 +1,84 @@
...
+- Rewrite modules into ES6 with classes ✏🛳
+- Move to a [monorepo](https://en.wikipedia.org/wiki/Monorepo) with separate packages ✏🛳
+- CLI tool ✏
+- Theme builder website 💡
+- Theme/plugin package sharing website 💡
+- Changing icon implementation from class based to HTML attribute based (`data-icon="user"`) ✏🛳
+- Theming implementation rewrite (behaviour will stay the same) ✏🛳
+- Move from LESS to SASS ✏🛳
+- v3 to v2 JavaScript pollyfill package 💡
+
```
Comment (lubber-de) [inline ROADMAP.md]: I suggest to add the following ideas for investigation (no promised features)
- A possible benefit when using CSS variables
```css
:root {
    --inverted-background: #1a1a1a;
}
div {
    background-color: var(--inverted-background);
}
```
- Support of IE11 only via 3rd party polyfills

File: ROADMAP.md [hunk 3]
```diff
@@ -1,34 +1,86 @@
...
+- Visual component tests ✏
+- CI/CD process ✏
+- Rewrite modules into ES6 with classes ✏🛳
+- Move to a [monorepo](https://en.wikipedia.org/wiki/Monorepo) with separate packages ✏🛳
+- CLI tool ✏
+- Theme builder website 💡
+- Theme/plugin package sharing website 💡
+- Changing icon implementation from class based to HTML attribute based (`data-icon="user"`) ✏🛳
+- Theming implementation rewrite (behaviour will stay the same) ✏🛳
+- Move from LESS to SASS ✏🛳
```
Comment (exoego) [inline ROADMAP.md]: What is a motivation to move to SASS?

Comment (hammy2899) [reply ROADMAP.md]: SASS is more favourable nowadays and it allows us to do a few tasks a bit nicer for a few things in v3 e.g. would be the `!default` expression which allows us to set default values which can help us make better theme support.

**Pull request merged**